server.go

```diff
@@ -249,8 +249,9 @@ func (s *Server) setClient(client MessagingClient) error {
 
 	// Start goroutine to read messages from the broker.
 	if client != nil {
-		s.done = make(chan struct{}, 0)
-		go s.processor(client, s.done)
+		done := make(chan struct{}, 0)
```

Comment:
Just me being paranoid. I like to pass in the local variable reference so there's no chance that `s.done` could change in between the instantiation and the passing of it as an arg. It should be used in a lock so that shouldn't happen but, again, I just get paranoid sometimes. :)